packages/storage/amplify_storage_s3/test/amplify_storage_s3_prefix_test.dart

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'dart:async';
+
+import 'package:amplify_core/amplify_core.dart';
+import 'package:amplify_storage_s3/method_channel_storage_s3.dart';
+import 'package:amplify_storage_s3/src/S3PrefixResolver/AmplifyStorageS3PrefixResolver.dart';
+import 'package:amplify_test/amplify_test.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+// Utilized from: https://github.com/flutter/flutter/issues/63465 #CyrilHu
+// For mocking Native -> Dart
+extension MockMethodChannel on MethodChannel {
+  Future<void> invokeMockMethod(String method, dynamic arguments) async {
+    const codec = StandardMethodCodec();
+    final data = codec.encodeMethodCall(MethodCall(method, arguments));
+
+    return ambiguate(ServicesBinding.instance)
+        ?.defaultBinaryMessenger
+        .handlePlatformMessage(
+          name,
+          data,
+          (ByteData? data) {},
+        );
+  }
+}
+
+void main() {
+  const MethodChannel storageChannel =
+      MethodChannel('com.amazonaws.amplify/storage_s3');
+
+  TestWidgetsFlutterBinding.ensureInitialized();
+
+  late StorageAccessLevel receivedAccessLevel;
+  String? receivedIdentityId;
+  String prefixResolver(
+      {required StorageAccessLevel storageAccessLevel, String? identityId}) {
+    receivedAccessLevel = storageAccessLevel;
+    receivedIdentityId = identityId;
+    return "";
+  }
+
+  setUp(() {
+    storageChannel.setMockMethodCallHandler((MethodCall methodCall) async {});
+    AmplifyStorageS3MethodChannel storage = AmplifyStorageS3MethodChannel(
+        prefixResolver: PrefixResolverTest(prefixResolver));
+    return storage.addPlugin();
+  });
+
+  test(
+      'PrefixResolver information from MethodChannel is properly serialized and called',
+      () async {
+    await storageChannel.invokeMockMethod("awsS3PluginPrefixResolver",
+        {'accessLevel': 'guest', 'targetIdentity': 'test-identity-id'});
+    expect(receivedAccessLevel, StorageAccessLevel.guest);
+    expect(receivedIdentityId, 'test-identity-id');
+  });
```

Comment:
Can we add some error tests, e.g. if `resolvePrefix` throws?

Reply:
Added please see updated code